sw/device/silicon_creator/lib/boot_svc/boot_svc_header.h

```diff
@@ -77,6 +78,20 @@ OT_ASSERT_SIZE(boot_svc_header_t, CHIP_BOOT_SVC_MSG_HEADER_SIZE);
 void boot_svc_header_finalize(uint32_t type, uint32_t length,
                               boot_svc_header_t *header);
 
+/**
+ * Checks the header of a boot services message.
+ *
+ * This function checks the identifier, digest, and length fields of a boot
+ * services message. Similar to the `boot_svc_header_finalize()` function above,
+ * this function also assumes that message payload starts immediately after the
+ * header and is exactly `length - sizeof(boot_svc_header_t)` bytes for digest
+ * computation.
+ *
+ * @param header Header of a boot services message.
+ * @return Whether the header is valid.
+ */
+rom_error_t boot_svc_header_check(const boot_svc_header_t *header);
```

Comment:
It might be surprising to readers that `boot_svc_header_check()` operates on the entire message rather than just the header. The same applies to `boot_svc_header_finalize()`.

Would you consider moving these functions into `boot_svc_msg.h` and operating on `const boot_svc_msg_t *` rather than reading more than `sizeof(boot_svc_header_t)` bytes through the header pointer?

Reply:
I agree. boot_svc_msg.h didn't exist when I initially defined these. I'll do this in a follow-up.